From 0ae0579f3c92ecf6270eea308905518ce75efb7b Mon Sep 17 00:00:00 2001
From: Jiffin Tony Thottan <jthottan@redhat.com>
Date: Wed, 22 Feb 2017 18:26:30 +0530
Subject: [PATCH 60/74] glusterd/ganesha : return proper value in pre_setup()

Change-Id: I6f7ce82488904c7d418ee078162f26f1ec81e9d9
Signed-off-by: Jiffin Tony Thottan <jthottan@redhat.com>
Reviewed-on: https://review.gluster.org/16733
Smoke: Gluster Build System <jenkins@build.gluster.org>
Reviewed-by: Atin Mukherjee <amukherj@redhat.com>
Reviewed-by: Raghavendra Talur <rtalur@redhat.com>
Tested-by: Raghavendra Talur <rtalur@redhat.com>
NetBSD-regression: NetBSD Build System <jenkins@build.gluster.org>
CentOS-regression: Gluster Build System <jenkins@build.gluster.org>
---
 xlators/mgmt/glusterd/src/glusterd-ganesha.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/xlators/mgmt/glusterd/src/glusterd-ganesha.c b/xlators/mgmt/glusterd/src/glusterd-ganesha.c
index 5d6144a..cd591df 100644
--- a/xlators/mgmt/glusterd/src/glusterd-ganesha.c
+++ b/xlators/mgmt/glusterd/src/glusterd-ganesha.c
@@ -834,15 +834,14 @@ pre_setup (gf_boolean_t run_setup, char **op_errstr)
 {
         int    ret = 0;
 
-        ret = check_host_list();
-
-        if (ret) {
+        if (check_host_list()) {
                 ret = setup_cluster(run_setup);
                 if (ret == -1)
                         gf_asprintf (op_errstr, "Failed to set up HA "
                                      "config for NFS-Ganesha. "
                                      "Please check the log file for details");
-        }
+        } else
+		ret = -1;
 
         return ret;
 }
-- 
1.8.3.1

